ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/StateMachine.hs

```diff
@@ -609,6 +638,11 @@ mockImpl model cmdErr = At <$> return mockResp
   where
     (mockResp, _dbModel') = step model cmdErr
 
+checkBlockIntegrity :: TestBlock -> Bool
+checkBlockIntegrity testBlock =
+  hashHeader (testHeader testBlock) == thHash (testHeader testBlock)
+  || not (tbIsValid (testBody testBlock))
+
```

Comment:
Isn't this the same as `Test.Ouroboros.Storage.TestBlock.testBlockIsValid`?

Reply:
Almost yes, this also checkes `tbIsValid`